### PR TITLE
opt: run page-cache commit and eviction on sync threadpool

### PR DESCRIPTION
**Pull request by rphmeier**
`PageCache::commit` has been split into two parts: `PageCache::prepare_transaction` and `PageCache::evict`.

This previously ran prior to sync and took a large portion of sync time, during which the disk was idle. Now, this runs in parallel with beatree's prepare_sync, and eviction has been moved out of the critical path altogether.

I measured a workload which took 310ms for `commit_and_prove` as of the parent commit. It's now down to 255ms, saving 55ms on average (which was typically what samply showed for the time of `PageCache::commit`).

In the same workload, the disk is now idle only during beatree `prepare_sync`. With future optimizations and interleaving, that gap should also disappear entirely.

### CODE REVIEW

**Comment by rphmeier** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/511?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#511** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/511?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#510** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/510?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#509** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/509?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rphmeier and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: nomt/src/page_cache.rs [hunk 1]
```diff
@@ -464,14 +464,12 @@ impl PageCache {
         }
     }
 
-    /// Flushes all the dirty pages into the underlying store.
+    /// Prepares a transaction of altered pages, according to the provided page diffs.
     /// This takes a read pass.
```
Comment (pepyakin) [inline nomt/src/page_cache.rs:468]: Does it actually take a read pass?

**Review by pepyakin — APPROVED** [review]

File: nomt/src/page_cache.rs (hunk 1 above)
Comment (rphmeier) [reply nomt/src/page_cache.rs:468]: Yes, see L474

**Comment by rphmeier** [comment]
### Merge activity

* **Oct 28, 11:30 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/nomt/511).

**Pull request merged**